gc/base/SweepHeapSectioning.cpp

```diff
@@ -458,3 +458,36 @@ MM_SweepHeapSectioning::reassignChunks(MM_EnvironmentBase *env)
 
 	return totalChunkCount;
 }
+
+/* For a given address determine the sweep chunk boundaries that contains it, we expect this chunk to be assigned during reassignChunks.*/
+void
+MM_SweepHeapSectioning::getChunkBoundariesForAddr(MM_EnvironmentBase *env,  void* addrToCheck, void* &sweepChunkBaseBoundary, void* &sweepChunkTopBoundary)
+{
+	MM_HeapRegionManager *regionManager = _extensions->getHeap()->getHeapRegionManager();
+	MM_HeapRegionDescriptor *region = regionManager->regionDescriptorForAddress(addrToCheck);
+
+	sweepChunkBaseBoundary = NULL;
+	sweepChunkTopBoundary = NULL;
+
+	uintptr_t *regionBase = (uintptr_t *)region->getLowAddress();
+	uintptr_t *regionTop = (uintptr_t *)region->getHighAddress();
+
+	Assert_MM_true(addrToCheck >= regionBase);
+	Assert_MM_true(addrToCheck < regionTop);
+
+	uintptr_t remainder = ((uintptr_t)addrToCheck - (uintptr_t)regionBase) % _extensions->parSweepChunkSize;
+	sweepChunkBaseBoundary = (void *) ((uintptr_t) addrToCheck - remainder);
+	sweepChunkTopBoundary = (void *) ((uintptr_t) sweepChunkBaseBoundary + _extensions->parSweepChunkSize);
+
+	/* Find out if the range of memory we are considering spans 2 different pools.  If it does,
+	 * the current chunk can only be attributed to one, so we limit the upper range of the chunk.
+	 */
+	void *poolHighAddr = NULL;
+	region->getSubSpace()->getMemoryPool(env, sweepChunkBaseBoundary, sweepChunkTopBoundary, poolHighAddr);
+	if (NULL == poolHighAddr) {
+		sweepChunkTopBoundary = (sweepChunkTopBoundary > regionTop ? regionTop : sweepChunkTopBoundary);
+	} else {
+		Assert_MM_true(poolHighAddr > sweepChunkBaseBoundary && poolHighAddr < sweepChunkTopBoundary);
```

Comment:
- can `poolHighAddr` be equal `sweepChunkTopBoundary` (and should condition be `<=`) ?
- would you please add extra brackets to not rely on order of operations?